src/go/rpk/pkg/config/redpanda_yaml.go

```diff
@@ -112,10 +112,12 @@ type (
 		AuthN   *string `yaml:"authentication_method,omitempty" json:"authentication_method,omitempty"`
 	}
 
+	// BACKCOMPAT 23-05-01: The CA used to be "truststore_file" in yaml; we
+	// deserialize truststore_file AND ca_file. See weak.go.
 	TLS struct {
```

Comment:
The struct isn't specific to clients, even though it's only used in a client right now. In fact looking at it ... this is a subset of fields in ServerTLS. Both of these seem to be named not the best. I err on keeping it for simplicity, since used elsewhere, `config.TLS` is less to see / read than `config.ClientTLS`. However it is only used in generate/prometheus.go and profile/create.go at the moment. Do you still prefer changing it?